nemo/collections/common/tokenizers/sentencepiece_tokenizer.py

```diff
@@ -0,0 +1,259 @@
+                tokens.append(self.tokenizer.id_to_piece(id))
+        return tokens
+
+    def add_special_tokens(self, special_tokens):
+        if isinstance(special_tokens, list):
+            for token in special_tokens:
+                if (
+                    self.tokenizer.piece_to_id(token) == self.tokenizer.unk_id()
+                    and token not in self.special_token_to_id
+                ):
+                    self.special_token_to_id[token] = self.vocab_size
+                    self.id_to_special_token[self.vocab_size] = token
+                    self.vocab_size += 1
+        elif isinstance(special_tokens, dict):
+            for token_name, token in special_tokens.items():
+                setattr(self, token_name, token)
+                if (
+                    self.tokenizer.piece_to_id(token) == self.tokenizer.unk_id()
+                    and token not in self.special_token_to_id
+                ):
+                    self.special_token_to_id[token] = self.vocab_size
+                    self.id_to_special_token[self.vocab_size] = token
+                    self.vocab_size += 1
+
+    @property
+    def pad_id(self):
+        return self.tokens_to_ids([self.pad_token])[0]
+
+    @property
+    def bos_id(self):
+        return self.tokens_to_ids([self.bos_token])[0]
+
+    @property
+    def eos_id(self):
+        return self.tokens_to_ids([self.eos_token])[0]
+
+    @property
+    def sep_id(self):
+        return self.tokens_to_ids([self.sep_token])[0]
+
+    @property
+    def cls_id(self):
+        return self.tokens_to_ids([self.cls_token])[0]
+
+
+def create_spt_model(
+    data_file: str,
+    vocab_size: int,
+    sample_size: int,
+    special_tokens: Union[Dict[str, str], List[str]],
+    do_lower_case: bool,
+):
+    """
+    Creates sentence piece tokenizer model from data file.
+    Args:
+        data_file: data file
+        vocab_size: vocabulary size
+        sample_size: maximum size of sentences the trainer loads
+        special_tokens: either list of special tokens or dictionary of token name to token value
+        do_lower_case: if text should be lower cased before tokenizer model is created
+    """
+    data_dir = os.path.dirname(data_file)
+    if special_tokens:
+        if isinstance(special_tokens, list):
+            special_tokens = list(set(special_tokens))
+        elif isinstance(special_tokens, dict):
+            special_tokens = list(set(special_tokens.values()))
+        vocab = special_tokens[:]
+    else:
+        vocab = []
+    output_dir = f'{data_dir}/spt'
+    if if_exist(output_dir, ['tokenizer.model']):
+        logging.info(f"tokenizer model {output_dir}/tokenizer.model already exists")
+        return f'{output_dir}/tokenizer.model', f'{output_dir}/vocab.txt'
+    logging.info(f'Processing WikiText dataset and store at {output_dir}')
+    os.makedirs(output_dir, exist_ok=True)
+
+    if not data_file:
+        files = glob.glob(f'{data_dir}/*.txt')
```

Comment:
I don't get this check, is it for None check or what? If data_file could be both path to the file and dir with files, could you please fix the args description? Could your please also add a check that the data_file as a file or dir exists

Reply:
i dont think this check is useful, deleted and added check for valid data_file above